UI/MapWnd.cpp

```diff
@@ -4925,33 +4911,20 @@ void MapWnd::CreateFleetButtonsOfType (
         for (int fleet_id : fleet_IDs)
             m_fleet_buttons[fleet_id] = fb;
 
-        AttachChild(fb);
         fb->LeftClickedSignal.connect(
-            boost::bind(&MapWnd::FleetButtonLeftClicked, this, fb));
+            boost::bind(&MapWnd::FleetButtonLeftClicked, this, fb.get()));
         fb->RightClickedSignal.connect(
-            boost::bind(&MapWnd::FleetButtonRightClicked, this, fb));
+            boost::bind(&MapWnd::FleetButtonRightClicked, this, fb.get()));
+        AttachChild(std::move(fb));
     }
 }
 
 void MapWnd::DeleteFleetButtons() {
+    for (auto& id_and_fb: m_fleet_buttons)
+        DetachChild(id_and_fb.second);
```

Comment:
why do you need to detach these but not the stationary or departing buttons before clearing their container?